testplan/report/testing/base.py

```diff
@@ -320,12 +274,27 @@ def runtime_status(self):
         return self._runtime_status
 
     @runtime_status.setter
-    def runtime_status(self, new_status):
-        """Set the runtime_status of all child entries."""
+    def runtime_status(self, new_status: str) -> None:
+        """
+        Sets the runtime status of all child entries.
+        """
         for entry in self:
             entry.runtime_status = new_status
         self._runtime_status = new_status
 
+    def set_runtime_status_filtered(
```

Comment:
This serves as an alternative setter, may replace entirely the existing one, but the behavior under such a change is not tested exhaustively. The purpose of the method is to work only with entries passed in addition to the new status.